Action commands

zahatikoff opened the pull request:
# What Does This PR Do?
This PR brings the basic action commands to the Sushi Bot. It also adds a small library for interaction with google Tenor API.

## Acceptance Criteria met
Which tasks will be completed with this PR?
- Sushi Bot now has a `bonk` command
- Sushi Bot now has a `hug` command
- Sushi Bot now has a `kiss` command

## Related Issues
Closes #4

zahatikoff committed:
#4 Add Axios dependency

zahatikoff committed:
#4 Add Tenor util library

Added a basic Tenor library that can provide the bot with gifs and
caches them for 12 hours.

zahatikoff committed:
#4 Add Tenor error type

zahatikoff committed:
refactor: Change from .then.catch to async/await

zahatikoff committed:
#4 Add action commands

ZacAndrei commented on src/util/errors.ts:
```diff
@@ -64,4 +64,10 @@ export class NotTextChannelError extends Error {
+
+export class TenorError extends Error {
+    public constructor() {
+        super("Tenor is unavailable or the request is malformed");
```
A new error type for when Tenor is unavailable or the request is malformed due to an API change

ZacAndrei replied:
A new error type for when Tenor is unavailable or the request is malformed due to an API change

ZacAndrei commented on src/util/tenor-utils.ts:
```diff
@@ -195,23 +195,20 @@ export class TenorSingleton {
-            // get a GIF's URL from response
-            const gifURL = gifObjects[i].media_formats.gif.url;
-            gifArray[i] = gifURL;
+        try {
```
Moved API request into a try/catch block, catch a failed request and rethrow as TenorError

ZacAndrei commented on package.json:
```diff
@@ -15,6 +15,7 @@
     "author": "Mirage Aegis",
     "license": "MIT",
     "dependencies": {
+        "axios": "^1.4.0",
```
added Axios as a dependency

zahatikoff replied:
Added axios as a dependency

zahatikoff commented on src/commands/action/pat.ts:
```diff
@@ -0,0 +1,69 @@
+            { name: "Format", value: `\`/${name} <target>\`` },
+            { name: "target", value: "Required parameter.The user you want to headpat" }
+        )
+};
```
Most commands can be simplified a lot with a builder, however i think that separating the files with some repetitions is better, because it clearly defines what the commands are

zahatikoff commented:
Man them pull requests be hard... Also i promise to actually schedule a lot more time for development in october as i go part-time
I think i also fucked up the conversations a bit bc idk why so yeahhh idk man

ZacAndrei commented:
And i think i fucked up the conversations 
Sorry man

MirageAegis commented on src/commands/action/pat.ts:
```diff
@@ -0,0 +1,69 @@
+        .setDescription("A command to headpat another user")
+        .setFields(
+            { name: "Format", value: `\`/${name} <target>\`` },
+            { name: "target", value: "Required parameter.The user you want to headpat" }
```
There seems to be a space missing after the period in all help embeds, could you fix that?

MirageAegis commented on src/commands/action/pat.ts:
```diff
@@ -0,0 +1,69 @@
+const name: string = "pat";
+
+export const command: Command = {
+    data: <SlashCommandBuilder>new SlashCommandBuilder()
```
I would also prefer having a space between the cast operation and the thing being casted while you're at it

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -0,0 +1,214 @@
+//
+
+const DEFAULT_RESPONSE_LIMIT = 100;
+const TWELVE_HOURS = 43200000;
```
Minor formatting request, separate every third power of 10 with an underscore

MirageAegis commented on src/util/tenor-utils.ts:
```diff
@@ -0,0 +1,214 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-present Zahatikoff
```
You can also change the copyright year of all the files you've worked on

MirageAegis reviewed (changes requested):
Looks good but there are a few changes I'd like. You should also put yourself in the license file

zahatikoff committed:
fix: Update year in license

zahatikoff committed:
fix: fix code formatting

Added spaces before casts and separated each 3rd power of 10 with an
underscore

zahatikoff committed:
fix: Fix typos.

Added spaces between sentences

zahatikoff committed:
Add myself to LICENSE

Is that what you wanted, Mira?
